### ISSUE TITLE
[bug] `useRequest.options.cacheKey` throw Uncaught Error when async error

### ISSUE THREAD

**Issue by Airkro** [alibaba/hooks#1420]
Version: v3.1.3

Example:

```jsx
import { render } from "react-dom";
import { useRequest } from "ahooks";

// ↓
async function fetching() {
  throw new Error("fail");
}

function Root() {
  useRequest(fetching, {
    cacheKey: "123",    // this line trigger error
    onError() {},       // not work
  });

  return null;
}

render(<Root />, document.querySelector("#root"));
```

相关： https://github.com/alibaba/hooks/issues/757

**Comment by brickspert** [comment]
Which version are you using? v2 or v3?

The v3 version has fixed this problem.

**Comment by Airkro** [comment]
Sorry, I have mentioned that now, v3

**Comment by brickspert** [comment]
fixed on v3.1.4
